lib/rspec/mocks/method_reference.rb

```diff
@@ -128,7 +130,7 @@ def method_implemented?(mod)
       #
       # This is necessary due to a bug in JRuby prior to 1.7.5 fixed in:
       # https://github.com/jruby/jruby/commit/99a0613fe29935150d76a9a1ee4cf2b4f63f4a27
-      if RUBY_PLATFORM == 'java' && JRUBY_VERSION.split('.')[-1].to_i < 5
+      if RSpec::Support::ComparableVersion.new(JRUBY_VERSION) < '1.7.5'
```

Comment:
You need the platform check or a `defined?` check as naturally `JRUBY_VERSION` is only defined on JRuby :)